packages/sources/lotus/src/index.ts

```diff
@@ -1,10 +1,13 @@
-import { expose } from '@chainlink/ea-bootstrap'
-import { makeExecute, endpointSelector } from './adapter'
-import * as endpoints from './endpoint'
-import { makeConfig, NAME } from './config'
-import * as types from './endpoint'
+import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
+import { config } from './config'
+import { balance } from './endpoint'
+import { PoRAdapter } from '@chainlink/external-adapter-framework/adapter/por'
 
-const adapterContext = { name: NAME }
+export const adapter = new PoRAdapter({
+  defaultEndpoint: balance.name,
+  name: 'LOTUS',
+  config,
+  endpoints: [balance],
+})
```

Comment:
Realized we may want a custom input validation here that returns a 400 error if the `addresses` input param array is of length 0. Looks like we have that validation in v2

Reply:
Agree, but i think this should be handled by the framework.  I'm not sure if [here](https://github.com/smartcontractkit/ea-framework-js/blob/main/src/validation/input-params.ts#L327) we need to check if the `length >=0`, instead of `length > 0`

Reply:
We may not want to add it to the framework in case an empty array is a valid input like in the case of `validatorStatus` in `eth-beacon`. Unless you were thinking about adding it just for `PoRBalanceEndpoint` and `PoRTotalBalanceEndpoint` specifically for the `addresses` input param. Otherwise this validation is specific to only a few adapters in which case we'd want `length > 0` ([example](https://github.com/smartcontractkit/external-adapters-js/blob/39d4cd3f1dc6072d082856da05cee5f3ecd9245c/packages/sources/lotus/src/endpoint/balance.ts#L49-L55)).

Reply:
Oh i see. Forgot the we have such case (empty array as a valid input param). I thought about having `customInputValidation` function in PoR adapters that will automatically check this condition but it will overwrite if such function exists on endpoint, so i think for now it would be better just to add  `customInputValidation` to every por balance EA.

Reply:
Sounds good to me 👍